config/config_params.go

```diff
@@ -467,6 +467,13 @@ func (config *Config) KubernetesProvider() Provider {
 	return ProviderNone
 }
 
+// ShouldEncryptWireguardHostTraffic factors in the kubernetes provider and config params to decide if host-traffic encryption should happen
+func (config *Config) ShouldEncryptWireguardHostTraffic() bool{
+	// currently, AKS *must* encrypt hostnetwork traffic, while it is optional on EKS
+	return config.KubernetesProvider() == ProviderAKS ||
+		(config.WireguardHostEncryptionEnabled && config.KubernetesProvider() == ProviderEKS)
```

Comment:
According to the [docs](https://docs.tigera.io/compliance/encrypt-cluster-pod-traffic), we support Wireguard on AKS with CalicoCNI (without host encryption), so I _think_ the if statement needs to be:
`return config.WireguardHostEncryptionEnabled &&
    (config.KubernetesProvider() == ProviderEKS || config.KubernetesProvider() == ProviderAKS)`

Reply:
Maybe we should just really lock host to host encryption down to the platforms we know we can support, i.e.
```
p := config.KubernetesProvider()
r := configParams.RouteSource
return config.WireguardHostEncryptionEnabled &&
  (p == ProviderAKS && r == "WorkloadIPs" ||
  p == ProviderEKS && r == "CalicoIPAM")`